app/javascript/components/widget-transition-controls.jsx

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect, useReducer } from 'react';
+import PropTypes from 'prop-types';
+import { mathMod } from 'ramda';
+
+const prefetchDelay = 1500;
+
+export const shouldRequestWidget = state =>
+  state.msRemainingSinceLastStart < Date.now() - state.startTimestamp;
+
+export const shouldRequestPrefetch = state =>
+  state.msRemainingSinceLastStart - (Date.now() - state.startTimestamp) <
+  prefetchDelay;
+
+export const isPaused = state => state.startTimestamp === null;
```

Comment:
nit: This pattern of thing, where you take the current state and you pull out a certain part is called a **selector**. We should call this `selectIsPaused`

Reply:
@chewnoill just implemented that change, and I also prepended 'select' to `shouldRequestPrefetch` and `shouldRequestWidget` as well for consistency.